.githooks/applypatch-msg

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+# Hook: applypatch-msg
+# This hook is invoked by git-am[1]. It takes a single parameter, the
+# name of the file that holds the proposed commit log message. Exiting
+# with a non-zero status causes git am to abort before applying the
+# patch.
+# 
+# The hook is allowed to edit the message file in place, and can be
+# used to normalize the message into some project standard format. It
+# can also be used to refuse the commit after inspecting the message
+# file.
+# 
+# The default applypatch-msg hook, when enabled, runs the commit-msg
+# hook, if the latter is enabled.
+#
+#### Sample file ####
+#
+# An example hook script to check the commit log message taken by
+# applypatch from an e-mail message.
+#
+# The hook should exit with non-zero status after issuing an
+# appropriate message if it wants to stop the commit.  The hook is
+# allowed to edit the commit message file.
+#
+# To enable this hook, rename this file to "applypatch-msg".
+
+#. git-sh-setup
+#commitmsg="$(git rev-parse --git-path hooks/commit-msg)"
+#test -x "$commitmsg" && exec "$commitmsg" ${1+"$@"}
+#:
+
+#### END sample file ####
+
+git_exec_path="$(git --exec-path)"
+
+if test -r "${git_exec_path}/git-sh-setup" ; then
+    . "${git_exec_path}/git-sh-setup" || exit 1
+fi
+
+require_work_tree_exists &&
```

Comment:
Doesn't the fact that this hook is run by git am mean that a working tree most be present?

Reply:
You can try `git am` in a bare repository, even if you should not.